app/policies/contact_policy.rb

```diff
@@ -0,0 +1,5 @@
+class ContactPolicy < ApplicationPolicy
+  def create?
+    Pundit.policy_scope!(user, Organization).present?
```

Comment:
This pattern gets used a lot in the other policies. Perhaps you should move it into `ApplicationPolicy` as a protected helper method?

That being said, I would write this policy method like so:

```ruby
def create?
  organization_policy.update?
end

protected

def organization_policy
  @organization_policy ||= Pundit.policy!(user, record.organization)
end
```

That way it's actually checking that the user creating the contact is able to update the organization that the contact belongs to. The current check is only checking that the user has access to any organization.

Reply:
I'm pretty sure the current check is working as expected. It's only allowing admins that belong to the organization. The test I added on line 75 of `organization_contacts_controller_spec.rb ` confirms this.

Having said that, this should be calling the `LocationPolicy` because that's what ultimately gets called to determine access, so might as well call it directly to avoid unnecessary DB queries.

Reply:
You were right! The test was not set up properly. I needed to create a location that the regular admin has access to, and then make sure they can only access stuff related to that location. After creating that location, the tests fail now. I will fix.